src/components/admin/UserProfileModal/UserProfileModal.tsx

```diff
@@ -0,0 +1,67 @@
+import { Fragment, useRef } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+
+import { STRING_ZERO_WIDTH_SPACE } from "settings";
+
+import { useProfileModalControls } from "hooks/useProfileModalControls";
+
+import { UserProfileModalBody } from "./components/UserProfileModalBody";
+
+export const UserProfileModal = () => {
+  const {
+    selectedUserId,
+    hasSelectedProfile,
+    closeUserProfileModal,
+  } = useProfileModalControls();
+
+  const cancelButtonRef = useRef(null);
```

Comment:
Do we need to add type to this hook?

Reply:
added: https://github.com/sparkletown/sparkle/pull/2931/commits/493bfda6c66a90eb3c85efbc5943a9e57dacef93